tool/generate.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/file_system/physical_file_system.dart';
+import 'package:analyzer/src/dart/sdk/sdk.dart';
+import 'package:analyzer/src/generated/engine.dart';
+import 'package:analyzer/src/generated/source.dart';
+import 'package:dart_style/dart_style.dart';
+import 'package:path/path.dart' as p;
+
+void main(List<String> args) {
+  var parser = new ArgParser();
+  parser.addFlag("dart1",
+      help: "Generate output for Dart 1.", negatable: false);
+  var options = parser.parse(args);
+
+  if (options.rest.length != 1) {
+    print("dart tool/generate.dart path/to/dart1/sdk");
+    print("");
+    print(parser.usage);
+    exitCode = 1;
+    return;
+  }
+
+  var generateDart1 = options["dart1"] as bool;
+  print("Generating constants for Dart ${generateDart1 ? 1 : 2}.");
+
+  new Directory("lib").createSync(recursive: true);
+
+  var dart1 = _createContext(options.rest.single);
+  var dart2 = _createContext(p.dirname(p.dirname(Platform.resolvedExecutable)));
+
+  for (var lib in dart1.sourceFactory.dartSdk.sdkLibraries) {
```

Comment:
Maybe consider an error message if the SDK directory isn't what's expected. Took me a couple of tries to get it right, and the only symptom of an invalid input is that it doesn't produce any output.